corehq/apps/data_dictionary/urls.py

```diff
@@ -15,7 +15,7 @@
 
 urlpatterns = [
     url(r"^json/$", data_dictionary_json, name='data_dictionary_json'),
-    url(r"^json/?(?P<case_type_name>\w+)/?$", data_dictionary_json, name='case_type_dictionary_json'),
+    url(r"^json/(?P<case_type_name>[\w-]+)/$", data_dictionary_json, name='case_type_dictionary_json'),
```

Comment:
oh interesting. what wrong urls would it have matched to earlier?

Reply:
"-" is a legal character in a case type name, and this change allows it. e.g. 

Currently in master:
```
>>> reverse('case_type_dictionary_json', kwargs={
...     'domain': 'domain',
...     'case_type_name': 'commcare-user',
... })
NoReverseMatch: Reverse for 'case_type_dictionary_json' with keyword arguments '{'domain': 'domain', 'case_type_name': 'commcare-user'}' not found. 1 pattern(s) tried: ['a/(?P<domain>[\\w\\.:-]+)/data_dictionary/json/?(?P<case_type_name>\\w+)/?$']
```

After this change:
```
>>> reverse('case_type_dictionary_json', kwargs={
...     'domain': 'domain',
...     'case_type_name': 'commcare-user',
... })
'/a/domain/data_dictionary/json/commcare-user/'
```

Reply:
P.S. The pattern currently in master also weirdly allows you to drop the forward slashes around the case type name. e.g.
```
>>> from django.urls import resolve
>>> resolve('/a/domain/data_dictionary/json/commcareuser/')  # ok
ResolverMatch(func=corehq.apps.data_dictionary.views.data_dictionary_json...
>>> resolve('/a/domain/data_dictionary/jsoncommcareuser')  # huh?
ResolverMatch(func=corehq.apps.data_dictionary.views.data_dictionary_json...
```

Reply:
@kaapstorm perfect catch of the reasons for this change.

@mkangia 
> Anyhow, that matter about trailing forward slash, I just hope no URL being formed in JS is currently dependent on that to work that way.

I can confirm this was not being used anywhere currently.